+++ switches.cc
@@ -185,9 +171,9 @@
 
   auto lookup_symbol = [&mapping, &size, symbol_prefix](
                            const fml::RefPtr<fml::NativeLibrary>& library) {
-    mapping = library->ResolveSymbol((symbol_prefix + "_start").c_str());
+    mapping = library->ResolveSymbol((symbol_prefix + "Data").c_str());
     size = reinterpret_cast<intptr_t>(
-        library->ResolveSymbol((symbol_prefix + "_size").c_str()));
+        library->ResolveSymbol((symbol_prefix + "Size").c_str()));
   };
 
   fml::RefPtr<fml::NativeLibrary> library =
@@ -328,13 +314,9 @@
       command_line.GetOptionValue(FlagForSwitch(Switch::ICUNativeLibPath),
                                   &native_lib_path);
 
-#if FML_OS_ANDROID
-      settings.icu_mapper = GetICUStaticMapping;
-#else
       settings.icu_mapper = [icu_symbol_prefix, native_lib_path] {
         return GetSymbolMapping(icu_symbol_prefix, native_lib_path);
       };
-#endif
     }
   }
 
